### PR TITLE
Auth enabled

### PR DESCRIPTION
**Pull request by darrenleeweber**
This branch enabled authentication and authorization based on trusted clients and access keys.  This is ready to go, although a couple of integration tests are pending for annotation creation and deletion.

### CODE REVIEW

**Comment by darrenleeweber** [comment]
To resolve conflict with master, used `git fetch origin; git rebase master` to replay `auth_enabled` commits on top of master (as described in http://stackoverflow.com/questions/7929369/how-to-rebase-local-branch-with-remote-master).  This fixed some `git rebase -i` work on the branch, which squashed some commits that were in master.

**Comment by darrenleeweber** [comment]
tagging @ndushay and @cbeer for review.  When merged with master, we plan to cut a new gem release and deploy to triannon dev.

**Comment by darrenleeweber** [comment]
:TODO:  Update README docs with authentication/authorization documentation
  - may require input on this by @azaroth42

**Comment by darrenleeweber** [comment]
@cbeer - coveralls indicates it has no data, it could be an occasional failure or a change in our configs, esp see the `spec/spec_helper.rb` where it has been modified to generate local data (available on dev-laptop) and it should also apply the coveralls formatter (see the 'SimpleCov Customization' section at https://coveralls.zendesk.com/hc/en-us/articles/201769485-Ruby-Rails).

**Comment by ndushay** [comment]
I think you need this line in spec_helper:

Coveralls.wear!

File: app/controllers/triannon/application_controller.rb [hunk 1]
```diff
@@ -39,13 +44,90 @@ def access_token_valid?(code)
...
+      end
+    end
+
+    # Map the request root container to config parameters;
+    # assume that a request can only map to one root container.
+    # If this mapping fails, assume that authorization is OK.
+    def request_container_config
+      # TODO: refine the matching algorithm, esp if there is more than one
+      # match rather than assume it can only match one container.
+      request_container = params['anno_root'] || request.path
```
Comment (azaroth42) [inline app/controllers/triannon/application_controller.rb]: Simplify to:
```
Triannon.config[:ldp]['anno_containers'][params['anno_root']]
```
Meaning... treat the anno_root as a key without trying to match beyond that. If it doesn't exist then it'll return nil, as expected?

**Pull request merged**